tests/realtikvtest/addindextest/integration_test.go

```diff
@@ -444,6 +444,26 @@ func TestAddIndexIngestCancel(t *testing.T) {
 	require.Empty(t, ingest.LitBackCtxMgr.Keys())
 }
 
+func TestAddIndexSplitTableRanges(t *testing.T) {
+	store := realtikvtest.CreateMockStoreAndSetup(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("drop database if exists addindexlit;")
+	tk.MustExec("create database addindexlit;")
+	tk.MustExec("use addindexlit;")
+	tk.MustExec(`set global tidb_ddl_enable_fast_reorg=on;`)
+
+	tk.MustExec("create table t (a int primary key, b int);")
+	for i := 0; i < 8; i++ {
+		tk.MustExec(fmt.Sprintf("insert into t values (%d, %d);", i*10000, i*10000))
+	}
+	tk.MustQuery("split table t between (0) and (80000) regions 7;").Check(testkit.Rows("6 1"))
+
+	ddl.SetBackfillTaskChanSizeForTest(4)
```

Comment:
Please add a case like `ddl.SetBackfillTaskChanSizeForTest(7)` which can test the logic of `len(remains)==0`.

Reply:
Rest LGTM